### PR TITLE
techdebt: [CDS-86988]: Troubleshoot Doc Updation 

### PR DESCRIPTION
**Pull request by SowmyaS63**
Troubleshooting option for repo connectivity to installed agent

### CODE REVIEW

File: docs/continuous-delivery/gitops/gitops-ref/troubleshooting.md [hunk 1]
```diff
@@ -124,3 +124,17 @@ This error indicates that the GitOps entity you are trying to create exists. It
 - A different scope (account, organization, or project) in Harness.
 
 - A different Argo CD project that is not mapped to a Harness project.
+  
+
+### Issue: "rpc error: code = Unknown desc = error testing repository connectivity: authorization failed"
+
+This error occured when attempted to connect a repo with installed agent and it's unable to connect repository in Github:
```
Comment (mankrit-singh) [inline docs/continuous-delivery/gitops/gitops-ref/troubleshooting.md:131]: This error occurs while attempting to connect a repository:

File: docs/continuous-delivery/gitops/gitops-ref/troubleshooting.md [hunk 2]
```diff
@@ -124,3 +124,17 @@ This error indicates that the GitOps entity you are trying to create exists. It
...
+
+- Make sure to monitor rate limitations on github account before attempting to connect repo.
```
Comment (mankrit-singh) [inline docs/continuous-delivery/gitops/gitops-ref/troubleshooting.md:133]: GitHub

File: docs/continuous-delivery/gitops/gitops-ref/troubleshooting.md [hunk 3]
```diff
@@ -124,3 +124,17 @@ This error indicates that the GitOps entity you are trying to create exists. It
...
+
+- Manage 'Rate Limiting' in Git by 'Enable Git Rate Limiting' to True.
```
Comment (mankrit-singh) [inline docs/continuous-delivery/gitops/gitops-ref/troubleshooting.md:135]: where is this setting in Git?

File: docs/continuous-delivery/gitops/gitops-ref/troubleshooting.md (hunk 1 above)
Comment (michaelcretzman) [inline docs/continuous-delivery/gitops/gitops-ref/troubleshooting.md:131]:
```suggestion
This error occurs when an agent is unable to connect to a repo:
```

File: docs/continuous-delivery/gitops/gitops-ref/troubleshooting.md (hunk 2 above)
Comment (michaelcretzman) [inline docs/continuous-delivery/gitops/gitops-ref/troubleshooting.md:133]:
```suggestion
- Ensure to review Github account rate limitations before attempting to connect to a repo.
```

File: docs/continuous-delivery/gitops/gitops-ref/troubleshooting.md (hunk 3 above)
Comment (michaelcretzman) [inline docs/continuous-delivery/gitops/gitops-ref/troubleshooting.md:135]:
```suggestion
- To manage rate limiting in GitHub, see [Enabling rate limits for Git](https://docs.github.com/en/enterprise-server@3.10/admin/configuration/configuring-user-applications-for-your-enterprise/configuring-rate-limits#enabling-rate-limits-for-git).
```

**Review by michaelcretzman — DISMISSED** [review]
Approved with edits I will incorporate

**Pull request merged**